modules/python/src2/typing_stubs_generation/api_refinement.py

```
@@ -93,19 +93,25 @@ def export_matrix_type_constants(root: NamespaceNode) -> None:
     )
 
 
-def make_optional_arg(arg_name: str) -> Callable[[NamespaceNode, SymbolName], None]:
+def make_optional_arg(*arg_names: str) -> Callable[[NamespaceNode, SymbolName], None]:
     def _make_optional_arg(root_node: NamespaceNode,
                            function_symbol_name: SymbolName) -> None:
         function = find_function_node(root_node, function_symbol_name)
         for overload in function.overloads:
-            arg_idx = _find_argument_index(overload.arguments, arg_name)
-            # Avoid multiplying optional qualification
-            if isinstance(overload.arguments[arg_idx].type_node, OptionalTypeNode):
-                continue
-
-            overload.arguments[arg_idx].type_node = OptionalTypeNode(
-                cast(TypeNode, overload.arguments[arg_idx].type_node)
-            )
+            for arg_name in arg_names:
```

It might sound controversial, but I would prefer outer loop over arguments rather than overloads.
Because it allows to add a check on the argument presence in at least 1 function overload instead of silently skip it (e.g. it will handle the "typo error" in argument name that is made optional).

Moreover, `_find_argument_index` should return `None` instead of rising the exception, because it is a valid case now.

---

That makes sense - in this context speed doesn't matter too much. Updated. 

Sample error for the 'typo error' case (wording pulled from the old `_find_argument_index` error)
> RuntimeError: Failed to find argument with name: 'distCoeff' in 'undistort' overloads